src/Str.php

```diff
@@ -55,6 +65,18 @@ public static function startsWith($subject, $start, $caseSensitive = true)
      */
     public static function symmetricSplit($subject, $delimiter, $limit, $default = null)
     {
+        if (empty($subject)) {
+            return [];
+        }
+
+        if ($delimiter === null) {
+            throw new LogicException('Required param "$delimiter" must not be null');
+        }
+
+        if ($limit === null) {
+            throw new LogicException('Required param "$limit" must not be null');
+        }
```

Comment:
Remove this. As soon as the type declarations get introduced this won't be necessary anymore, but the exception is different. Though, I don't think this happens *accidentally*. The delimiter and limit are rarely chosen on demand, but are rather nearly always hard-coded values.